### PR TITLE
Storage transmission evaluator

### PR DESCRIPTION
**Pull request by vishweshbankwar**
This PR covers the algorithm to evaluate whether an exporter can transmit files from storage. The algorithm is based on following idea.

Consider exporter received total of 5 consecutive batches in time window of T and each of those exports took E1, E2, E3, E4, E5 time respectively
 
```text
|--------------------------------------T-------------------------------------------|
|--E1--|
                |--E2--|
                                 |--E3--|
                                                           |--E4--|
                                                                            |--E5--|
```
The above representation shows that

1) Exporter was busy exporting average of **E** = (E1+E2+E3+E4+E5)/5 time.

2) Exporter was idle for average of **I** = (T-E)/5 time.

The average idle time is the time interval between each export which can be used to export files from storage. So for any export we can calculate how many files can be transmitted from storage as follows:
```text 
(I - E)/E 
```

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/StorageTransmissionEvaluator.cs [hunk 1]
```diff
@@ -12,7 +13,7 @@ internal class StorageTransmissionEvaluator
         private double[] _exportIntervalsInSeconds;
         private int _transmissionDurationIndex = -1;
         private int _exportIntervalIndex = -1;
-        private DateTime prevExportTime;
+        private long prevExportTime;
```
Comment (reyang) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/StorageTransmissionEvaluator.cs:16]: Consider making this explicit, e.g. `prevExportTimestampTicks`

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/StorageTransmissionEvaluator.cs [hunk 2]
```diff
@@ -41,8 +42,11 @@ internal void UpdateTransmissionDuration(double currentExportDuration)
 
         internal void UpdateExportInterval()
         {
+            long currentTime = Stopwatch.GetTimestamp();
             // todo: check if this can fail
-            double exportInterval = (DateTime.UtcNow - prevExportTime).TotalSeconds;
+            double exportInterval = TimeSpan.FromTicks(currentTime - prevExportTime).TotalSeconds;
```
Comment (reyang) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/StorageTransmissionEvaluator.cs:47]: Consider explicit name e.g. `exportIntervalSeconds`

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/StorageTransmissionEvaluator.cs [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            _sampleSize = sampleSize;
+            _transmissionDurationsInSeconds = new double[sampleSize];
+            _exportIntervalsInSeconds = new double[sampleSize];
+        }
+
+        internal void UpdateTransmissionDuration(double currentExportDuration)
+        {
+            _transmissionDurationIndex++;
+            // if we run out of elements, start from beginning
+            if (_transmissionDurationIndex == _transmissionDurationsInSeconds.Length)
```
[the remark this answers is not in the document]
Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/StorageTransmissionEvaluator.cs:33]: Yes - missed to update it here. will do

**Review by rajkumar-rangaraj — APPROVED** [review]
LGTM

**Pull request merged**